src/goto-instrument/cover.cpp

```diff
@@ -1061,15 +1196,25 @@ void instrument_cover_goals(
   if(ignore_trivial && program_is_trivial(goto_program))
     return;
 
-  const namespacet ns(symbol_table);
-  basic_blockst basic_blocks(goto_program);
-  std::set<unsigned> blocks_done;
-
   // ignore if built-in library
   if(!goto_program.instructions.empty() &&
-     goto_program.instructions.front().source_location.is_built_in())
+     (goto_program.instructions.front().source_location.is_built_in() ||
+      // 'built-in' for Java
+      (has_prefix(
+         id2string(goto_program.instructions.front().function),
+         "java::array") ||
+       (has_prefix(
+          id2string(goto_program.instructions.front().function),
+          "java::java.lang") &&
```

Comment:
I suggest to add a command line option similar to `--java-cp-include-files` that allows to specify the scope of instrumentation.